dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtensions.cs

```diff
@@ -83,6 +83,25 @@ public static partial class PromptyKernelExtensions
         Verify.NotNull(kernel);
         Verify.NotNullOrWhiteSpace(promptyTemplate);
 
+        var promptTemplateConfig = ToPromptTemplateConfig(promptyTemplate);
+
+        return KernelFunctionFactory.CreateFromPrompt(
+            promptTemplateConfig,
+            promptTemplateFactory ?? s_defaultTemplateFactory,
+            kernel.LoggerFactory);
+    }
+
+    /// <summary>
+    /// Create a <see cref="PromptTemplateConfig"/> from a prompty template.
+    /// </summary>
+    /// <param name="promptyTemplate">Prompty representation of a prompt-based <see cref="KernelFunction"/>.</param>
+    /// <returns>The created <see cref="PromptTemplateConfig"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="promptyTemplate"/> is null.</exception>
+    /// <exception cref="ArgumentException"><paramref name="promptyTemplate"/> is empty or composed entirely of whitespace.</exception>
+    public static PromptTemplateConfig ToPromptTemplateConfig(string promptyTemplate)
```

Comment:
Isn't it a bit weird to have a public non-extension method here?  As a user I wouldn't expect to have to call a function directly on an extension class.

Reply:
Have you considered to instead add an extension method on the `IPromptTemplateFactory` called `CreateFromPromptyTemplate(string template)`?

Reply:
I'll follow the pattern in the Functions.Yaml package